### ISSUE TITLE
ARC abrupt closure

### ISSUE THREAD

**Issue by AnushaSeetharaman** [jarrodek/ChromeRestClient#894]
Your description here

Hello team,

ARC has been updated few days back. The Histroy is displayed in a well organized manner in this version. The drawback what I face is that, it get closed often every 10 minutes if its left idle. 
"Getting error as a pop up message "......, Click on the ballon to restart ARC""

It is irritating to often open ARC each and every 15 mins. 

Since my project uses Spring and MVC, I have to test the services using ARC often. It will be helpful if you could take care of this issue. 

Old version I never faced this problem, and ARC will be opened for the whole day. 
The same problem is also faced by my team mates. 

## Expected outcome
What should happen?
Require ARC window to be opened unless I click on the close button.
## Actual outcome
What happened?
Often the window gets closed with an error message pop up.
## Versions
App: 8.14.55.288-stable
Chrome: 54.0.2840.99
Platform: Windows (7)

## Steps to reproduce
1. Open ARC and hit an URL
2. Leave the window unsued for atleast 15 mins.
3. It gets closed automatically

**Comment by jarrodek** [comment]
Hi,

Thank you for the issue report.
It might be related to issue #893. Other ARC user reported that they may be a memory leak issue in the app. I tried to reproduce it and I can see huge memory usage during run time but so far no luck with pointing the script that causes it. 

One if causes of app crashes in Chrome is when the app consumes too much of memory.  However, when the app is idle it is not doing anything and therefor it should not use more memory. 
You can try to test it by running task manager (Chrome options dropdown > More tools > Task manager) and observe memory consumption while idle. It shouldn't change. 

I will try to fix it asap but like I said, so far no luck with finding the code that may cause an issue.

Thank you for your patience.

**Comment by jpkrebs** [comment]
This issue continues for me and renders the client unusable.  Once ARC crashes due to this issue, requests sent using the "send" button never return; the only way I can make it respond again is by uninstalling and reinstalling ARC or rebooting.

**Comment by jarrodek** [comment]
@jpkrebs you say the window is closing after pressing the SEND button or it is loading all the time and the response never come back? If the second then can you please check logs if there's any error in it? Or can you paste version number of your app.

**Comment by jpkrebs** [comment]
@jarrodek When I press send, it looks like it's working, but the call is never actually made, and eventually it times out.  The remote server is never contacted.  Meanwhile memory used (according to Chrome task manager) is shooting up the whole time, looks like about 32MB per second; the memory usage keeps rising, even after the call times out, and eventually ARC crashes.  There is nothing in the log file, only the "reusing old log" message.  The version is 8.19.60.296.  Is there any way to roll back to an older version?  It used to work great until a few weeks ago.

**Comment by jpkrebs** [comment]
I have a bit more info to report on this issue.  I used a packet sniffer and discovered that as soon as I start ARC, it sends hundreds of requests per second that look like this:

CONNECT www.google-analytics.com:443 HTTP/1.1
Host: www.google-analytics.com:443
Proxy-Connection: keep-alive
User-Agent: Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/54.0.2840.99 Safari/537.36

The firewall I'm behind blocks www.google-analytics.com, so the calls are all erroring back.  When I'm no longer subject to this firewall rule, it works fine.  Is there any way to fix this so the error occurs once and doesn't keep trying?

**Comment by jarrodek** [comment]
Thanks @jpkrebs for more details. It's strange behavior. It shouldn't affect the app though. GA implementation is using Fetch API meanwhile the app works directly on socket. 

Can you try disable GA and try again if this helps? You can disable GA in settings.

**Comment by jpkrebs** [comment]
Yes, disabling GA in the settings makes it behave properly even behind the firewall.  Thanks for the tip!  One minor thing-- it seems to always set itself back to "on" when I restart ARC.  That is easy to work around though.

**Comment by jarrodek** [comment]
It shouldn't do this. It stores its state and should restore it when the app is opened. I need to check this as well.
